h5py/_hl/files.py

```diff
@@ -567,6 +570,53 @@ def __init__(self, name, mode='r', driver=None, libver=None, userblock_size=None
 
         super().__init__(fid)
 
+    _in_memory_file_counter = 0
+
+    @classmethod
+    @with_phil
+    def in_memory(cls, file_image=None, **kwargs):
+        """Create an HDF5 file in memory, without an underlying file
+
+        file_image
+            The initial file contents as bytes (or anything that supports the
+            Python buffer interface). HDF5 takes a copy of this data.
+        block_size
+            Chunk size for new memory alloactions (default 64 KiB).
+
+        Other keyword arguments are like File(), although name, mode,
+        driver and locking can't be passed.
+        """
+        for k in ('driver', 'locking', 'backing_store'):
+            if k in kwargs:
+                raise TypeError(
+                    f"File.in_memory() got an unexpected keyword argument {k!r}"
+                )
+        fcpl_kwargs = {}
+        for k in inspect.signature(make_fcpl).parameters:
+            if k in kwargs:
+                fcpl_kwargs = kwargs.pop(k)
```

Comment:
```suggestion
                fcpl_kwargs[k] = kwargs.pop(k)
```

Reply:
Thanks, good catch. I'll ensure this is covered in tests at the same time.